### PR TITLE
Add a cache for the code generators

### PR DESCRIPTION
**Pull request by prajwalkkumar**
This commit adds functionality to cache the code generators for offline processing.

### CODE REVIEW

File: servicex/app/codegen.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        url: Optional[str] = url_cli_option,
+        backend: Optional[str] = backend_cli_option,
+        flush: Optional[bool] = typer.Option(None,
+                                             "--flush",
+                                             help="Flush the codegens from the cache"
+                                             )):
+    """
+    List the available code generators
+    """
+    sx = ServiceXClient(url=url, backend=backend)
```
Comment (BenGalewsky) [inline servicex/app/codegen.py:50]: This should use the cache! I tried switching off wifi and this call didn't work

File: servicex/app/codegen.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+from servicex.app.cli_options import url_cli_option, backend_cli_option
+from servicex.servicex_client import ServiceXClient
+from typing import Optional
+
+codegen_app = typer.Typer(name="codegen", no_args_is_help=True)
+
+
+@codegen_app.command(no_args_is_help=False)
+def codegens(
```
Comment (BenGalewsky) [inline servicex/app/codegen.py:40]: This winds up with a weird command line
```
servicex codegen codegens --flush
```

Maybe 
```
servicex codegens flush
and 
servicex codegens list
```

**Review by BenGalewsky — CHANGES_REQUESTED** [review]
Weird CLI user experience - `servicex codegen codegens` doesn't make sense
I tried 
`servicex codegen codegens --flush`
then
` servicex codegen codegens` prints out 
```
Returning code generators from cache
```
They should not be in the cache at that point!

The examples are now working - try these out and make sure this all works as expected. Turn off your wifi and make sure you can get results out of cache if you are on an airplane

File: servicex/app/codegen.py (hunk 1 above)
Comment (prajwalkkumar) [reply servicex/app/codegen.py:59]: This was using the cache. It works now with wifi off after some more changes.

File: servicex/app/codegen.py (hunk 2 above)
Comment (prajwalkkumar) [reply servicex/app/codegen.py]: This has the expected cli now. 
I moved it out of main.py and hence we were seeing this behaviour. 
servicex codegens flush
servicex codegens list work now

**Comment by prajwalkkumar** [comment]
Made the changes requested!

**Pull request merged**